### PR TITLE
Filters &  Modals

### PR DESCRIPTION
**Pull request by MarkVed17**
- Functionalities added:
    - Post
        - Edit Post
        - Delete Post
        
    - Comment
        - Edit Comment
        - Delete Comment
        
    - Profile
        - Edit Profile
        - Followers List
        - Following List

    - Filters
        - Feed Page
        - Explore Page

### CODE REVIEW

**Review by horsemaker — APPROVED** [review]
Looks good Vedant! 👌

**Review by Radheshyam1234 — COMMENTED** [review]
All looks good, just one issue is that on opening the followers/following list a modal opens containing userProfile card and when you click on those cards you are redirected to that user profile which is good but Modal still remains open  and it shows followers/following details of that user which is obviously not good.

File: src/frontend/components/CommentDropdownMenu.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  const deleteCommentHandler = async () => {
+    const response = await dispatch(
+      deletePostComment({ postId, commentId, token })
+    );
+    if (response?.payload?.posts !== undefined) {
+      toast.success("Your comment has been deleted successfully!", {
+        style: {
+          background: "#22c55e",
+          color: "#FFFFFF",
```
Comment (t007rushi) [inline src/frontend/components/CommentDropdownMenu.js:24]: styling is common you can export out and import for every time

File: src/frontend/components/ProfileSection.js [hunk 2]
```diff
@@ -1,7 +1,17 @@
-import React from "react";
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { followUser, unfollowUser } from "../features";
+import { EditProfileModal } from "./EditProfileModal";
+import { FollowersUsersModal } from "./FollowersUsersModal";
+import { FollowingUsersModal } from "./FollowingUsersModal";
```
Comment (t007rushi) [inline src/frontend/components/ProfileSection.js:6]: barrel  import

File: src/frontend/components/ProfileSection.js [hunk 3]
```diff
@@ -12,76 +22,133 @@ const ProfileSection = ({ profile }) => {
...
-            </p>
-            <p className="text-sm text-gray-500 dark:text-gray-400">
-              @{username}
-            </p>
+    <>
+      <div className="flex p-2 gap-5">
+        <img
+          className="w-24 h-24 rounded-full sm:w-28 sm:h-28"
+          src={
+            avatarURL === null
```
Comment (t007rushi) [inline src/frontend/components/ProfileSection.js:43]: can ```!avatarURL``` works ?

**Review by t007rushi — APPROVED** [review]
Loos good to go ahead

**Comment by MarkVed17** [comment]
> All looks good, just one issue is that on opening the followers/following list a modal opens containing userProfile card and when you click on those cards you are redirected to that user profile which is good but Modal still remains open  and it shows followers/following details of that user which is obviously not good.

Yeah, thanks for pointing this out. I'll make the necessary changes. Do let me know if anything else comes up.

File: src/frontend/components/CommentDropdownMenu.js (hunk 1 above)
Comment (MarkVed17) [reply src/frontend/components/CommentDropdownMenu.js:24]: Yeah, could do that.

**Pull request merged**